src/Common/PageCache.cpp

```diff
@@ -145,9 +145,10 @@ bool PageCache::contains(const PageCacheKey & key, bool inject_eviction) const
     return shard.contains(key_hash);
 }
 
-void PageCache::Shard::onRemoveOverflowWeightLoss(size_t weight_loss)
+void PageCache::Shard::onValueRemoval(const MappedPtr & mappedPtr)
 {
-    ProfileEvents::increment(ProfileEvents::PageCacheWeightLost, weight_loss);
+    ProfileEvents::increment(ProfileEvents::PageCacheWeightLost);
```

Comment:
Need to pass `cell.size` here along with `cell.value` and increment profile event by this size, otherwise this place is incorrect (increment by 1 is not what we want here).

Reply:
Please check [comment below](https://github.com/ClickHouse/ClickHouse/pull/80799#discussion_r2135914535) for a clarification on this.